File: listview-java/app/src/main/java/com/bmarques/listview/ui/activity/OrderListActivity.java
package com.bmarques.listview.ui.activity;

import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.bmarques.listview.R;
import com.bmarques.listview.dao.OrderDao;
import com.bmarques.listview.model.Order;

public class OrderListActivity extends AppCompatActivity {

    public static final String APP_BAR_TITLE = "Order List";
    private final OrderDao dao = new OrderDao();
    private ArrayAdapter<Order> adapter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_order_list);
        setTitle(APP_BAR_TITLE);
        configureList();

        inputOrdersMock();

    }

    private void inputOrdersMock() {
        dao.save(new Order("Tiago", 50D));
        dao.save(new Order("Samira", 10D));
        dao.save(new Order("Nicolas", 10D));
        dao.save(new Order("Valentina", 10D));
        dao.save(new Order("Carlos", 10D));
        dao.save(new Order("Roseli", 10D));
        dao.save(new Order("Alessandra", 10D));
        dao.save(new Order("Rafael", 10D));
        dao.save(new Order("André", 10D));
        dao.save(new Order("Daiane", 10D));
        dao.save(new Order("Caio", 10D));
        dao.save(new Order("Letícia", 10D));
        dao.save(new Order("Lívia", 10D));
        dao.save(new Order("Kelly", 10D));
        dao.save(new Order("Maikel", 10D));
        dao.save(new Order("Jair", 10D));
        dao.save(new Order("Leonete", 10D));
    }

    @Override
    protected void onResume() {
        super.onResume();
        atualizaAlunos();
    }

    private void atualizaAlunos() {
        adapter.clear();
        adapter.addAll(dao.findAll());
    }

    private void configureList() {
        ListView listaDeAlunos = findViewById(R.id.activity_order_list_listview);
        configuraAdapter(listaDeAlunos);
    }

    private void configuraAdapter(ListView orderListView) {
        adapter = new ArrayAdapter<>(
                this,
                android.R.layout.simple_list_item_1);
        orderListView.setAdapter(adapter);
    }
}

File: kotlin/ListViewKotlin/app/src/main/java/com/bmarques/listviewkotlin/ui/activity/OrderListActivity.kt
package com.bmarques.listviewkotlin.ui.activity

import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import com.bmarques.listviewkotlin.R
import com.bmarques.listviewkotlin.dao.OrderDao
import com.bmarques.listviewkotlin.model.Order

class OrderListActivity : AppCompatActivity() {
    private val dao: OrderDao = OrderDao()
    private var adapter: ArrayAdapter<Order>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_order_list)
        title = APP_BAR_TITLE
        configureList()
        inputOrdersMock()
    }

    private fun inputOrdersMock() {

        var order = Order("Tiago", 50.0)

        dao.save(Order("Tiago", 50.0,))
        dao.save(Order("Samira", 10.0,))
        dao.save(Order("Nicolas", 10.0,))
        dao.save(Order("Valentina", 10.0,))
        dao.save(Order("Carlos", 10.0,))
        dao.save(Order("Roseli", 10.0,))
        dao.save(Order("Alessandra", 10.0,))
        dao.save(Order("Rafael", 10.0,))
        dao.save(Order("André", 10.0,))
        dao.save(Order("Daiane", 10.0,))
        dao.save(Order("Caio", 10.0,))
        dao.save(Order("Letícia", 10.0,))
        dao.save(Order("Lívia", 10.0,))
        dao.save(Order("Kelly", 10.0,))
        dao.save(Order("Maikel", 10.0,))
        dao.save(Order("Jair", 10.0,))
        dao.save(Order("Leonete", 10.0,))
    }

    override fun onResume() {
        super.onResume()
        atualizaAlunos()
    }

    private fun atualizaAlunos() {
        adapter!!.clear()
        adapter!!.addAll(dao.findAll())
    }

    private fun configureList() {
        val listaDeAlunos = findViewById<ListView>(R.id.activity_order_list_listview)
        configuraAdapter(listaDeAlunos)
    }

    private fun configuraAdapter(orderListView: ListView) {
        adapter = ArrayAdapter<Order>(
            this,
            android.R.layout.simple_list_item_1
        )
        orderListView.adapter = adapter
    }

    companion object {
        const val APP_BAR_TITLE = "Order List - Kotlin"
    }
}
